internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -709,6 +648,12 @@ func getControllerDeploymentVersion(deployment *appsv1.Deployment) (string, erro
 }
 
 func processDeploymentHealth(deployment *appsv1.Deployment) (bool, string, string) {
+
+	if deployment == nil {
+		msg := "Numaflow Controller Deployment not found"
+		return false, "Progressing", msg
+	}
```

Comment:
this is related to [this](https://github.com/numaproj/numaplane/pull/125/files#r1697647636) comment